Proxy client move#16513 keichwa (#1364)
uyuni-project/uyuni-docs: modules/client-configuration/pages/client-proxy.adoc

A list of chained proxies for a Salt client can be seen by clicking the name of the client in menu:Systems[All], then select the [guimenu]``Connection`` subtab of the [guimenu]``Details`` tab.

== Move Clients between Proxies

You can move Salt and Salt SSH push clients between proxies without the need to repeat the registration process.

[NOTE]
====

> If you want to move a traditional client between proxies or any client between chained proxies you must repeat the registration process from the beginning.

Author: ```suggestion
If you want to move a traditional client between proxies you must repeat the registration process from the beginning.
```

> ====

Reviewer: This is not correct.
Moving salt client between chained proxies is supported.
Moving chained proxy server is not supported even if it is managed by salt. Proxy server must always registered from the beginning.

Author: ok, thanks.  I got it wrong.  I think we can simplify it.

// Procede as follows:

.Procedure: Moving Salt or  Salt SSH Push Client between Proxies

. In the {productname} {webui}, navigate to the [guimenu]``System Details`` page for the client you want to move between proxies.

> . Navigate to the [guimenu]``Connection`` tab.

Reviewer: Then follow "Change proxy" link to see the drop-down menu.

> . Navigate to the [guimenu]``Connection`` tab.

Author: ```suggestion
. Navigate to the [guimenu]``Connection`` tab.
  Then follow the [guimenu]``Change proxy`` link to see the drop-down menu.
```

. From the [guimenu]``New Proxy`` drop-down menu select the proxy you want the clients to move to, and click btn:[Change Proxy].

The same functionality is also available with the [systemitem]``system.changeProxy`` API call.

=== Background Information
The effect of this function differs between normal Salt clients and Salt SSH push clients.

==== Normal Salt Clients

The function schedules a Salt state action, which modifies [literal]``master:`` setting in the [path]``susemanager.conf`` Salt client configuration file to point to the new proxy.
Then the function restarts the Salt client.

[NOTE]
====
Changing [literal]``master:`` by manually editing the [path]``susemanager.conf`` file has the same effect and is supported, too.
====

When the minion restarts and reconnects via the new proxy, the server updates the proxy path in the database and schedules another action for refreshing the channel URLs.

==== Salt SSH Push Clients

The function updates the proxy path immediately in the database and new action
for refreshing the channel URLs is scheduled.

== Move Clients from Proxies to the Server

> If you want to move any Salt or traditional client from proxies to the server you must repeat the registration process from the beginning.

Author: ```suggestion
If you want to move a Salt client from a proxy to the server, select [literal]``None`` from proxy list.

If you want to move a traditional client to the server you must repeat the registration process from the beginning.
```

> 

Reviewer: Moving salt client from proxy to server is supported, just select "None" from proxy list.

Author: ok
